AzureRedisEvents.md

```diff
@@ -1,143 +1,81 @@
+
+With sufficient resilience, applications gracefully handle any brief connection loss or cache unavailability like that experienced during node maintenance. It’s also possible that your application might unexpectedly lose its connection to the cache without warning from AzureRedisEvents because of network errors or other events.
+
+The AzureRedisEvents channel isn't a mechanism that can notify you days or hours in advance. Maintenance events aren't intended to allow time for a person to be alerted and take manual action, and most of them will be fired within minutes or seconds of maintenance.
+
+We only recommend subscribing to AzureRedisEvents in a few noteworthy cases:
+
+* Applications with extreme performance requirements, where even minor delays must be avoided. In such scenarios, traffic could be seamlessly rerouted to a backup cache before maintenance begins on the current cache.
+* Applications that explicitly read data from replica rather than primary nodes. During maintenance on a replica node, the application could temporarily switch to read data from primary nodes.
+* Applications that can't risk write operations failing silently or succeeding without confirmation, which can happen as connections are being closed for maintenance. If those cases would result in dangerous data loss, the application can proactively pause or redirect write commands before the maintenance is scheduled to begin.
+
+## Types of events
+
+AzureRedisEvents currently sends the following notifications:
+* `NodeMaintenanceScheduled`: Indicates that a maintenance event is scheduled. Usually between 10-15 minutes in advance.
+* `NodeMaintenanceStarting`: Fired ~20s before maintenance begins
+* `NodeMaintenanceStart`: Fired when maintenance is imminent (<5s)
+* `NodeMaintenanceFailoverComplete`: Indicates that a replica has been promoted to primary
+* `NodeMaintenanceEnded`: Indicates that the node maintenance operation is over
+
+## Subscribing to the channel
+
+If you are using C#, we recommend relying on the StackExchange.Redis library. It will automatically subscribe to the pub/sub channel and raise the ServerMaintenanceEvent in response to maintenance. For more information, please see the [StackExchange.Redis documentation](https://github.com/StackExchange/StackExchange.Redis/blob/main/docs/ServerMaintenanceEvent.md).
+
+For other client libraries, you will need to subscribe to the AzureRedisEvents pub/sub channel and implement logic to parse incoming maintenance event messages.
 
 ## Message format
 
-Each entry in the message is separated by using a pipe (|) as a delimiter. All messages start with a *FieldName* followed by the *Entry* for the field and so on so forth.
+Messages sent by Azure Redis are pipe (|) delimited strings. All messages start with a *FieldName* followed by the *Entry* for the field and so on so forth.

```

Comment:
"All messages start with a *FieldName* followed by the *Entry* for the field, followed by additional pairs of field names and entries"